### DISCUSSION TITLE
[Chapter05] 매개변수 줄이기, 범용처리 클래스 관련 질문

### DISCUSSION THREAD

**Discussion by snoopingh** [cowaine/is-my-code-that-strange#36] [Chapter 04-05 (불변, 응집도)] [answered]
1. 매개변수는 몇 개 정도가 적당하고, 줄인다면 어떤 방법으로 하고 계시나요?
2. 범용 처리 클래스 (Common / Util) 작성시에 본인만의 규칙 같은 것이 있으실까요?

**Comment by firewoody237** [comment, marked as answer]
클린코드에서는 0개가 가장 이상적이라곤 하는데,
저는 최대 4개까지 쓰고(대신 네이밍은 가능한 확실하게..) 그것보다 많아지면 클래스를 따로 만드려고는 하는데,
4개를 넘어간적은 크게 없었던 것 같아요.

---

`static`으로 쓸법하고, 인스턴스 변수가 필요 없고, 2군데 이상에서 쓰면 주로 빼는 것 같습니다.
다른 조건들이 만족되도 2~3군데 이상에서 쓰지 않으면 따로 안빼는 것 같아요.

@snoopingh 님의 방식도 같이 말씀주시면 좋겠네요 ㅎㅎ

**Comment by corock** [comment]
> Q1. 매개변수는 몇 개 정도가 적당하고, 줄인다면 어떤 방법으로 하고 계시나요?

이건 정해져 있는 게 아니라 개발자나 팀 환경에 따라 다를 것 같습니다. 다만, 이펙티브 자바 [아이템 51](https://github.com/orgs/Study-2-Effective-Java/discussions/131) 에 관련 내용이 나오긴 합니다.
저는 보통 4개 이상이 되면 요청(`XxxRequest`) 클래스를 만들고 매개변수 목록을 필드에 부여합니다.

> Q2. 범용 처리 클래스 (Common / Util) 작성시에 본인만의 규칙 같은 것이 있으실까요?

- `Common` 문자열이 들어간 형식으로는 이 책에서 처음 봤고, 프로젝트를 하면서 만든 적은 없습니다.
- `Util` 클래스는 여태껏 종종 만들었던 것 같은데 이 책을 읽고 이제 웬만하면 만들지 않으려고 합니다. 이전에는 주로 네이밍에 민감했습니다.
  - 클래스 이름은 복수형으로 짓는다. (_e.g._ `OpenSearchUtils`)
  - 패키지 이름은 단수형으로 짓는다. (_e.g._ `com.**.util.OpenSearchUtils`)

**Comment by coalong** [comment]
2. 범용 처리 클래스 (Common / Util) 작성시에 본인만의 규칙 같은 것이 있으실까요?

-> 추가 질문으로 Common 이나 Util 같은 클래스를 실무에서 사용하는 경우, 어떤 예시가 있을까요?
